src/main/java/teammates/ui/controller/InstructorFeedbackEditPageData.java

```diff
@@ -12,68 +12,235 @@
+                                                                        isRecipientNameVisibleFor);
+        qnForm.setVisibilitySettings(visibilitySettings);
+        
+        qnForm.setQuestionHasResponses(questionHasResponses.get(question.getId()));
+        
+        qnForm.setQuestionSpecificEditFormHtml(question.getQuestionDetails().getQuestionSpecificEditFormHtml(question.questionNumber));
+        qnForm.setEditable(false);
+        
+        qnForms.add(qnForm);
+    }
+
+
+    private void buildNewQuestionForm(FeedbackSessionAttributes feedbackSession, int nextQnNum) {
+      
+        Url doneEditingLink = new Url(Const.ActionURIs.INSTRUCTOR_FEEDBACKS_PAGE)
+                                .withUserId(account.googleId)
+                                .withCourseId(feedbackSession.courseId)
+                                .withSessionName(feedbackSession.feedbackSessionName);
+        newQnForm = FeedbackQuestionEditForm.getNewQnForm(doneEditingLink.toString(), feedbackSession,
+                                                          getQuestionTypeChoiceOptions(), getParticipantOptions(null, true),
+                                                          getParticipantOptions(null, false), getQuestionNumberOptions(nextQnNum), 
+                                                          getNewQuestionSpecificEditFormHtml());
+    }
+
     /**
      * Returns a list of HTML options for selecting participant type.
      * Used in instructorFeedbackEdit.jsp for selecting the participant type for a new question.
-     * isGiver refers to the feedback path (!isGiver == feedback's target)
+     * isGiver refers to the feedback path (!isGiver == feedback's recipient)
      */
-    public List<String> getParticipantOptions(FeedbackQuestionAttributes question, boolean isGiver) {
-        List<String> result = new ArrayList<String>();
+    private List<ElementTag> getParticipantOptions(FeedbackQuestionAttributes question, boolean isGiver) {
+        List<ElementTag> result = new ArrayList<ElementTag>();
         for (FeedbackParticipantType option : FeedbackParticipantType.values()) {
+            
             boolean isValidGiver = isGiver && option.isValidGiver();
             boolean isValidRecipient = !isGiver && option.isValidRecipient();
             
             if (isValidGiver || isValidRecipient) {
-                String optionStr = "<option value=\"%s\"%s>%s</option>";
                 String participantName = isValidGiver ? option.toDisplayGiverName()
                                                       : option.toDisplayRecipientName();
                 
-                String selected = "";
+                boolean isSelected = false;
                 // for existing questions
                 if (question != null) {
-                    boolean isGiverType = isValidGiver && question.giverType == option;
+                    boolean isGiverType     = isValidGiver     && question.giverType == option;
```

Comment:
We don't seem to do vertical alignment usually... but in any case, if you're doing it, keep it consistent within this method at least.